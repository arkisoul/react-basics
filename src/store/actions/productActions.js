import axios from 'axios';

export const ADDPRODUCT = "ADDPRODUCT";
export const REMOVEPRODUCT = "REMOVEPRODUCT";
export const PRODUCT_REQUEST_START = "PRODUCT_REQUEST_START";
export const PRODUCT_REQUEST_SUCCESS = "PRODUCT_REQUEST_SUCCESS";
export const PRODUCT_REQUEST_FAILURE = "PRODUCT_REQUEST_FAILURE";

export const addProduct = (product) => ({
  type: ADDPRODUCT,
  payload: { product: product },
});
export const removeProduct = (productId) => ({
  type: REMOVEPRODUCT,
  payload: { productId: productId },
});
export const productFetchStart = () => ({ type: PRODUCT_REQUEST_START });
export const productFetchSuccess = (products) => ({
  type: PRODUCT_REQUEST_SUCCESS,
  payload: { products },
});
export const productFetchFailure = (error) => ({
  type: PRODUCT_REQUEST_FAILURE,
  payload: { error },
});

export const fetchProducts = () => async (dispatch) => {
  try {
    dispatch(productFetchStart());
    const res = await axios.get("https://fakestoreapi.com/products");
    const products = res.data;
    dispatch(productFetchSuccess(products))
  } catch (error) {
    dispatch(productFetchFailure(error))
  }
};
