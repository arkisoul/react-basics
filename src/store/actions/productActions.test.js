import { expect, it, describe } from "@jest/globals";

import * as productActions from "./productActions";

describe("product actions", () => {
  const product = {
    id: 1,
    title: "Product title",
    body: "Product description",
    price: 12.99,
  };

  it("should return add product action object", () => {
    expect(productActions.addProduct(product)).toEqual({
      type: productActions.ADDPRODUCT,
      payload: { product },
    });
    expect(productActions.addProduct(product)).toMatchSnapshot();
  });
  
  it("should return remove product action object", () => {
    expect(productActions.removeProduct(product.id)).toEqual({
      type: productActions.REMOVEPRODUCT,
      payload: { productId: product.id },
    });
    expect(productActions.removeProduct(product.id)).toMatchSnapshot();
  });
});
