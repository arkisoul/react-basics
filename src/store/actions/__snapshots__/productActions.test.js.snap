// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product actions should return add product action object 1`] = `
Object {
  "payload": Object {
    "product": Object {
      "body": "Product description",
      "id": 1,
      "price": 12.99,
      "title": "Product title",
    },
  },
  "type": "ADDPRODUCT",
}
`;

exports[`product actions should return remove product action object 1`] = `
Object {
  "payload": Object {
    "productId": 1,
  },
  "type": "REMOVEPRODUCT",
}
`;
